src/main/java/ch/njol/skript/expressions/ExprColorOf.java

```diff
@@ -113,76 +172,119 @@ public Class<?>[] acceptChange(ChangeMode mode) {
 		if (FireworkEffect.class.isAssignableFrom(returnType))
 			return CollectionUtils.array(Color[].class);
 
+		switch (mode) { // items/blocks/entities don't accept these change modes
+			case ADD:
+			case REMOVE:
+			case REMOVE_ALL:
+				return null;
+		}
+
 		if (mode != ChangeMode.SET && !getExpr().isSingle())
 			return null;
 
-		if (Entity.class.isAssignableFrom(returnType))
-			return CollectionUtils.array(Color.class);
-		else if (Block.class.isAssignableFrom(returnType))
-			return CollectionUtils.array(Color.class);
-		if (ItemType.class.isAssignableFrom(returnType))
-			return CollectionUtils.array(Color.class);
-		return null;
+		return CollectionUtils.array(Color.class);
```

Comment:
I don't really like the use of blanket acceptance for modes instead of explicitly stating which are allowed, but I don't want to block the PR over it. If you agree it's worth fixing, go for it; if you don't, that's fine too.